test/unit/python/sample_files/write_init_files/physics_inputs_constituent_dim.F90

```diff
@@ -0,0 +1,404 @@
+         if (len_trim(missing_required_vars) > 0) then
+            call endrun("Required variables missing from registered list of input variables: "//&
+               trim(missing_required_vars))
+         end if
+
+         ! End simulation if there are protected input variables that are not initialized:
+         if (len_trim(protected_non_init_vars) > 0) then
+            call endrun("Required, protected input variables are not initialized: "//&
+               trim(protected_non_init_vars))
+         end if
+
+         ! Deallocate required variables array for use in next suite:
+         deallocate(ccpp_required_data)
+
+      end do !CCPP suites
+
+      ! Read in constituent variables if not using init variables
+      field_data_ptr => cam_constituents_array()
+
+      ! Iterate over all registered constituents
+      do constituent_idx = 1, size(const_props)
+         var_found = .false.
+         ! Check if constituent standard name in registered SIMA standard names list:
+         call const_props(constituent_idx)%standard_name(std_name)
+         if(any(phys_var_stdnames == trim(std_name))) then
+            ! Don't read the variable in if it's already initialized
+            if (is_initialized(std_name)) then
+               cycle
+            end if
+            ! Find array index to extract correct input names:
+            do n=1, phys_var_num
+               if(trim(phys_var_stdnames(n)) == trim(std_name)) then
+                  const_input_idx = n
+                  exit
+               end if
+            end do
+            call read_field(file, std_name, input_var_names(:,const_input_idx), 'lev', timestep, field_data_ptr(:,:,constituent_idx),                  &
+                 mark_as_read=.false., error_on_not_found=.false., var_found=var_found)
+         else
+            ! If not in standard names list, then just use constituent name as input file name:
+            call read_field(file, std_name, [std_name], 'lev', timestep, field_data_ptr(:,:,constituent_idx), mark_as_read=.false.,                    &
+                 error_on_not_found=.false., var_found=var_found)
+         end if
+         if(.not. var_found) then
+            constituent_has_default = .false.
+            call const_props(constituent_idx)%has_default(constituent_has_default, constituent_errflg, constituent_errmsg)
+            if (constituent_errflg /= 0) then
+               call endrun(constituent_errmsg, file=__FILE__, line=__LINE__)
+            end if
+            if (.not. constituent_has_default) then
+               ! Intialize to constituent's configured minimum value
+               call const_props(constituent_idx)%minimum(constituent_min_value, constituent_errflg, constituent_errmsg)
+               field_data_ptr(:,:,constituent_idx) = constituent_min_value
+               if (masterproc) then
+                  write(iulog,*) 'Constituent ', trim(std_name), ' default value not configured.  Setting to 0.'
```

Comment:
I realize this isn't directly related to your changes, but it would probably be good to modify the code generator to log the actual `constituent_min_value` here instead of just assuming that it is zero:

```suggestion
                  write(iulog,*) 'Constituent ', trim(std_name), ' default value not configured.  Setting to min value of ', constituent_min_value
```